plugin/discovery/find.go

```diff
@@ -0,0 +1,167 @@
+package discovery
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"runtime"
+	"strings"
+)
+
+const machineName = runtime.GOOS + "-" + runtime.GOARCH
```

Comment:
This was sloppy reading of the RFC combined with the fact that we use dashes as the separators for the filenames within the directory. I'll change it. :+1: